python_docs_theme/layout.html

```diff
@@ -45,7 +45,7 @@
 
 {% block footer %}
     <div class="footer">
-    &copy; <a href="{{ pathto('copyright') }}">{% trans %}Copyright{% endtrans %}</a> {{ copyright|e }}.
+    &copy; <a href="https://github.com/python/devguide/blob/master/LICENSE">{% trans %}Copyright{% endtrans %}</a> {{ copyright|e }}.
```

Comment:
The python-docs-theme is a common library, and used not just by devguide but also other projects, like [Python Packaging](https://packaging.python.org/) and [pylint](http://pylint.pycqa.org/en/latest/).
 
So perhaps we need a way to customize this.

Reply:
Random contributor stopping by, but what about a theme variable (with the default being a URL to the canonical PSF license)?

Reply:
Having a theme variable sounds great, so it can be configured for each project.

Reply:
Ah yes, that would make much more sense! @CAM-Gerlach Thanks for the suggestion. @Mariatta Thanks for the feedback. @CAM-Gerlach I'm less familiar with Sphinx themes, so I likely won't have a solution right away. If you'd like to propose specific changes, then you are more than welcome to.

Reply:
```suggestion
    &copy; <a href="{{ theme_copyright_url }}">{% trans %}Copyright{% endtrans %}</a> {{ copyright|e }}.
```

Sure, you'd just apply the above change and add the following line to the `theme.conf` (using the current de-facto value as the default, for backwards compat):

```ini
copyright_url = https://docs.python.org/copyright.html
```

And then any theme user that wanted to customize the link could do so in the `html_theme_options` dict in the `conf.py`, i.e.

```python
html_theme_options = {
    "copyright_url": "https://www.example.com",
}
```

Lmk if I can be of any further assistance!

Reply:
[Here's](https://github.com/spyder-ide/spyder-docs-sphinx-theme/commit/5f9fe5d90a16b8b15261ef4eacd53a9040fd539e) a relatively minimal example of adding a footer link, if that's helpful.

Reply:
```suggestion
    &copy; {% if theme_copyright_url or hasdoc('copyright') %}<a href="{% if theme_copyright_url %}{{ theme_copyright_url }}{% else %}{{ pathto('copyright') }}{% endif %}">{% endif %}{% trans %}Copyright{% endtrans %}{% if theme_copyright_url or hasdoc('copyright') %}</a>{% endif %} {{ copyright|e }}.
```

Here's a better solution that is both more backwards-compatible, more functional and does the "right thing" more of the time:

* If the new `copyright_url` theme option is defined, uses that,
* Otherwise, if a file named `copyright` exists in the repo, links to that (current behavior, fully backward compatible)
* Otherwise, doesn't make copyright a link at all, so it isn't broken

The would be coupled with the following in the `theme.conf`:

```ini
copyright_url =
```

This would be fully backwards-compatible and make more sense especially for non-PSF projects, so they wouldn't point to your license by default that doesn't apply to them.

Reply:
@CAM-Gerlach  Thanks for taking the time to explain this. Since you kindly asked if you could be of more assistance, there is something I'd like to understand more clearly.

Using a theme variable as you suggested sounds like the best option here, but I'm a little fuzzy on how to leave options open for all users of the python-docs-theme library while also setting a specific option for when Python's devguide is using that library:

- If I define a `copyright_url` theme option in `theme.conf`, then I'm making changes to the library rather than to how the devguide is using the library, so that doesn't seem like a viable option.

- If I create a `copyright` file in the python-docs-theme library, then I have the same problem. I wondered if you might have meant creating a `copyright.rst` in the devguide repo, but that would mean having the same information in multiple places, which isn't ideal.

As a result, the cleanest solution seems to me to be the first one you suggested, where you:

- Add a theme variable to `layout.html` with no conditionals
- Add a default URL to `theme.conf`
- Specify which URL the devguide will be using for the copyright by adding a key-value pair to the dictionary `html_theme_options` inside the devguide's `conf.py`

As I type all this it, it occurs to me that maybe you were proposing that we add a theme variable with the conditionals you suggested *and* add the URL to the key-value pair to the dictionary `html_theme_options` inside the devguide's `conf.py`. Is that what you meant?

Reply:
> As I type all this out, it occurs to me that maybe you were proposing that we add a theme variable with the conditionals you suggested _and_ add the URL to the key-value pair to the dictionary `html_theme_options` inside the devguide's `conf.py`. Is that what you meant?

Yup, exactly; I just didn't explicitly specify that again, sorry. My revised proposal is essentially a superset of the original; it still allows consumers to set a URL (or local path) of their choice in their `conf.py` just like the previous (i.e. if `copyright_url` is specified in `conf.py` theme options, the behavior is identical), while behaving much better if it is absent:

* Keeping the current behavior by default if `copyright.rst` is present, so themes can still use their existing local `copyright.rst` file if they prefer, without having explicitly change their config and hardcode the path
* In case that file is missing and no link is specified, don't generate a broken link (to fix the proximate bug in python/devguide#652 and python/devguide#653 for all theme users automatically, without any further user action)
* Not automatically changing the URL for all existing users to point to Python's own copyright statement, unless they take explicit action to specify their own (avoiding any unintended legal uncertainty for either users or the PSF)

This does mean PSF projects that don't include their own `copyright.rst` and want to point to the standard Python copyright statement will need to add one line to their `html_theme_options` with the link, but I would think that relatively small one-time cost is less than that of affecting backward compat and defaulting to a potentially legally problematic value for non-PSF users, and still fixes the immediate bug (broken link) without users having to do anything.